### PR TITLE
(deprecated) cosmic-player: init at unstable-2024-03-13 

### PR DESCRIPTION
**Pull request by sofiedotcafe**
## Description of changes

Media Player for the COSMIC Desktop Environment.

Please note, the project is still heavily work in progress, so expect it not being so finished and stable!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/co/cosmic-player/package.nix [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+, just
+, pkg-config
+, libxkbcommon
+, wayland
+, xorg
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "cosmic-player";
+  version = "unstable-2024-06-10";
```
Comment (donovanglover) [inline pkgs/by-name/co/cosmic-player/package.nix:19]:
```suggestion
  version = "0-unstable-2024-06-10";
```
[The `version` attribute must start with a digit](https://github.com/NixOS/nixpkgs/tree/master/pkgs#package-naming)

File: pkgs/by-name/co/cosmic-player/package.nix [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+  src = fetchFromGitHub {
+    owner = "pop-os";
+    repo = pname;
```
Comment (donovanglover) [inline pkgs/by-name/co/cosmic-player/package.nix:23]:
```suggestion
    repo = "cosmic-player";
```
[Avoid `repo = pname`](https://github.com/nix-community/nixpkgs-lint/issues/21)

File: pkgs/by-name/co/cosmic-player/package.nix [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+          xorg.libXcursor
+          xorg.libXrandr
+          xorg.libXi
+          wayland
+          libxkbcommon
+        ]
+      }
+  '';
+
+  meta = with lib; {
```
Comment (donovanglover) [inline pkgs/by-name/co/cosmic-player/package.nix:89]:
```suggestion
  meta = {
```
[Avoid `with lib;`](https://github.com/NixOS/nixpkgs/issues/208242)

**Review by donovanglover — COMMENTED** [review]
Welcome to nixpkgs! Make sure to format your code with [`nixfmt-rfc-style`](https://search.nixos.org/packages?query=nixfmt-rfc) available in [`shell.nix`](https://wiki.nixos.org/wiki/Development_environment_with_nix-shell)

**Comment by jh-devv** [comment]
> Welcome to nixpkgs! Make sure to format your code with [`nixfmt-rfc-style`](https://search.nixos.org/packages?query=nixfmt-rfc) available in [`shell.nix`](https://wiki.nixos.org/wiki/Development_environment_with_nix-shell)

Thanks and well yeah... I forgot to do that :3

File: pkgs/by-name/co/cosmic-player/package.nix [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+          wayland
+          libxkbcommon
+        ]
+      }
+  '';
+
+  meta = {
+    homepage = "https://github.com/pop-os/cosmic-player";
+    description = "Media Player for the COSMIC Desktop Environment";
+    license = licenses.gpl3Only;
```
Comment (Aleksanaa) [inline pkgs/by-name/co/cosmic-player/package.nix:92]:
```suggestion
    license = lib.licenses.gpl3Only;
```

File: pkgs/by-name/co/cosmic-player/package.nix [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+    maintainers = with maintainers; [ jh-devv ];
```
Comment (Aleksanaa) [inline pkgs/by-name/co/cosmic-player/package.nix:93]:
```suggestion
    maintainers = with lib.maintainers; [ jh-devv ];
```

File: pkgs/by-name/co/cosmic-player/package.nix [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    platforms = platforms.linux;
```
Comment (Aleksanaa) [inline pkgs/by-name/co/cosmic-player/package.nix:94]:
```suggestion
    platforms = lib.platforms.linux;
```

File: pkgs/by-name/co/cosmic-player/package.nix [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        "sha256-/7twYMt5/LpzxLXAQKTGNnWcfspUkkZsN5hJu7KaANc=";
+      "smithay-clipboard-0.8.0" =
+        "sha256-MqzynFCZvzVg9/Ry/zrbH5R6//erlZV+nmQ2St63Wnc=";
+      "taffy-0.3.11" = "sha256-SCx9GEIJjWdoNVyq+RZAGn0N71qraKZxf9ZWhvyzLaI=";
+      "winit-0.29.10" = "sha256-ScTII2AzK3SC8MVeASZ9jhVWsEaGrSQ2BnApTxgfxK4=";
+    };
+  };
+
+  postPatch = ''
+    substituteInPlace justfile --replace '#!/usr/bin/env' "#!$(command -v env)"
```
Comment (Aleksanaa) [inline pkgs/by-name/co/cosmic-player/package.nix:55]: Did't see this, but there's a prefix https://github.com/pop-os/cosmic-player/blob/52b9439ca4ff4d2daeefc18ea5ba90cc8c36886c/justfile#L5

File: pkgs/by-name/co/cosmic-player/package.nix [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  nativeBuildInputs =
+    [ just pkg-config makeBinaryWrapper rustPlatform.bindgenHook ];
+  buildInputs = [ wayland libxkbcommon alsa-lib ffmpeg llvmPackages.clang ];
+
+  dontUseJustBuild = true;
+
+  justFlags = [
+    "--set"
+    "prefix"
+    (placeholder "out")
```
Comment (Aleksanaa) [inline pkgs/by-name/co/cosmic-player/package.nix:67]:
```suggestion
    "--set prefix ${placeholder "out"}"
```

Also for the other

File: pkgs/by-name/co/cosmic-player/package.nix (hunk 7 above)
Comment (jh-devv) [reply pkgs/by-name/co/cosmic-player/package.nix:55]: I mean, I can replace it in the justfile too but doesn't the arguments passed to the just script already force that?

Comment (Aleksanaa) [reply pkgs/by-name/co/cosmic-player/package.nix:55]: Yeah, that's enough. Just saying this may not be used.

**Comment by ahoneybun** [comment]
@jh-devv I think you just need to use `nixfmt-rfc-style` to format the package.nix file then commit that to fix the last issue with the CI.

**Review by camerondugan — CHANGES_REQUESTED** [review]
Great work, just needs some formatting to pass a test

**Review by ahoneybun — APPROVED** [review]
This looks good now!

**Review by camerondugan — APPROVED** [review]
Looks good, I compiled and played an mp4

**Comment by jh-devv** [comment]
Silly me :3 Accidentally deleted the branch lol, it's back up now!

**Comment by Aleksanaa** [comment]
Try updating version and switch to different ffmpeg version?

**Comment by jh-devv** [comment]
> Silly me :3 Accidentally deleted the branch lol, it's back up now!

Did this again! Can't revert since not at home 😭

**Comment by jh-devv** [comment]
New PR: #350710

**Pull request closed without merging**